packages/sources/coinmetrics/src/transport/price-ws.ts

```diff
@@ -60,6 +63,16 @@ export const calculateAssetMetricsUrl = (
   desiredSubs: (typeof assetMetricsInputParameters.validated)[],
 ): string => {
   const { API_KEY, WS_API_ENDPOINT } = context.adapterSettings
+
+  //remove assets from desiredSubs that are invalid
+  const updated_desiredSubs: (typeof assetMetricsInputParameters.validated)[] = []
+  desiredSubs.forEach((pair) => {
+    if (!invalid_base_currencies.includes(pair['base'].toLowerCase())) {
+      updated_desiredSubs.push(pair)
+    }
+  })
+  desiredSubs = updated_desiredSubs
```

Comment:
Could replace this with a `filter`: 
```suggestion
  desiredSubs = desiredSubs.filter(({base}) => !invalid_base_currencies.includes(base.toLowerCase()))
```